### PR TITLE
remove on field when ingest result and pass in dataset id when doing upload_result, allow chunk processing

### PR DESCRIPTION
**Pull request by grant-Kolena**
### Linked issue(s):
blocked by this https://github.com/kolenaIO/shipyard/pull/1737
### Please check if the PR fulfills these requirements

- [ ] Include reference to internal ticket and/or GitHub issue "Fixes #NNNN" (if applicable)
- [ ] Relevant tests for the changes have been added
- [ ] Relevant docs have been added / updated

### CODE REVIEW

**Comment by grant-Kolena** [comment]
Integ test should fail due to [this](https://github.com/kolenaIO/shipyard/pull/1737) is not in

File: kolena/_experimental/dataset/_evaluation.py [hunk 1]
```diff
@@ -216,27 +169,29 @@ def test(
...
+        if isinstance(df_result_input, pd.DataFrame):
+            _validate_datapoint_id_columns(df_result_input, existing_dataset.id_fields)
+            df_results = _process_result(config, df_result_input, existing_dataset.id_fields)
+            upload_data_frame(df=df_results, batch_size=BatchSize.UPLOAD_RECORDS.value, load_uuid=load_uuid)
+        else:
+            id_column_validated = False
+            for df_result in df_result_input:
+                if not id_column_validated:
+                    _validate_datapoint_id_columns(df_result, existing_dataset.id_fields)
+                    id_column_validated = True
```
Comment (ly-kolena) [inline kolena/_experimental/dataset/_evaluation.py:189]: Nit: theoretically, df_result in def_result_input could be of different shape

File: kolena/_experimental/dataset/_evaluation.py [hunk 2]
```diff
@@ -216,27 +169,29 @@ def test(
...
     :param results: Either a DataFrame or a list of tuples, where each tuple consists of
                     a eval configuration and a DataFrame.
-    :param on: The column(s) to merge on between datapoint DataFrame and result DataFrame
-
     :return None
     """
-    df_data = _fetch_dataset(dataset)
-    df_datapoints = _to_deserialized_dataframe(df_data, column=COL_DATAPOINT)
-    log.info(f"fetched {len(df_data)} for dataset {dataset}")
+    existing_dataset = load_dataset(dataset)
```
Comment (ly-kolena) [inline kolena/_experimental/dataset/_evaluation.py:169]: check for None?

File: kolena/_experimental/dataset/_evaluation.py [hunk 3]
```diff
@@ -76,58 +77,34 @@ def _fetch_results(dataset: str, model: str) -> pd.DataFrame:
...
-        )
+    df_serialized_datapoint_id_object = _to_serialized_dataframe(
+        df_result[sorted(id_fields)],
+        column=COL_DATAPOINT_ID_OBJECT,
     )
-    return df_results
+    df_result_eval = _to_serialized_dataframe(df_result.drop(columns=id_fields), column=COL_RESULT)
+    df_result_eval[COL_EVAL_CONFIG] = json.dumps(eval_config) if eval_config is not None else None
+    df_result_eval = pd.concat([df_result_eval, df_serialized_datapoint_id_object], axis=1)
+    df_result_eval["datapoint_id"] = -1
```
Comment (ly-kolena) [inline kolena/_experimental/dataset/_evaluation.py]: can be removed

File: tests/integration/_experimental/dataset/test_evaluation.py [hunk 4]
```diff
@@ -148,6 +173,48 @@ def test__test__multiple_eval_configs() -> None:
     assert fetched_eval_config_2 == eval_config_2
     expected_df_result_1 = df_result_1.drop(columns=[JOIN_COLUMN])[3:10].reset_index(drop=True)
     expected_df_result_2 = df_result_2.drop(columns=[JOIN_COLUMN])[3:10].reset_index(drop=True)
+    result_columns_1.remove(JOIN_COLUMN)
```
Comment (ly-kolena) [inline tests/integration/_experimental/dataset/test_evaluation.py]: I think we should restore id columns also for result. We dropped it behind users' back, when we return it to the users, it should look like the inputs they gave us.

cc @yifanwu-kolena

Comment (ly-kolena) [reply tests/integration/_experimental/dataset/test_evaluation.py]: This is existing behavior, so punting to future discussion if there's user feedback.

File: kolena/_experimental/dataset/_evaluation.py (hunk 3 above)
Comment (grant-Kolena) [reply kolena/_experimental/dataset/_evaluation.py]: will do

File: kolena/_experimental/dataset/_evaluation.py (hunk 1 above)
Comment (grant-Kolena) [reply kolena/_experimental/dataset/_evaluation.py:189]: the number of columns should be the same right?

File: tests/integration/_experimental/dataset/test_evaluation.py [hunk 5]
```diff
@@ -69,12 +70,37 @@ def test__test() -> None:
...
+    assert eval_cfg is None
+    expected_df_dp = df_dp[3:10].reset_index(drop=True)
+    expected_df_result = df_result.drop(columns=[JOIN_COLUMN])[3:10].reset_index(drop=True)
+    _assert_frame_equal(fetched_df_dp, expected_df_dp, dp_columns)
+    _assert_frame_equal(fetched_df_result, expected_df_result, result_columns)
+
+
+def test__test__iterator_input() -> None:
+    dataset_name = with_test_prefix(f"{__file__}::test__test")
+    model_name = with_test_prefix(f"{__file__}::test__test")
```
Comment (yifanwu-kolena) [inline tests/integration/_experimental/dataset/test_evaluation.py]: names should be consistent with func name

Comment (grant-Kolena) [reply tests/integration/_experimental/dataset/test_evaluation.py]: will do

File: kolena/_experimental/dataset/_evaluation.py [hunk 6]
```diff
@@ -216,27 +168,31 @@ def test(
...
-    :param on: The column(s) to merge on between datapoint DataFrame and result DataFrame
-
     :return None
     """
-    df_data = _fetch_dataset(dataset)
-    df_datapoints = _to_deserialized_dataframe(df_data, column=COL_DATAPOINT)
-    log.info(f"fetched {len(df_data)} for dataset {dataset}")
+    existing_dataset = load_dataset(dataset)
+    if not existing_dataset:
+        raise IncorrectUsageError(f"dataset {dataset} does not exist")
```
Comment (ly-kolena) [inline kolena/_experimental/dataset/_evaluation.py]: Woult `NotFoundError` be better?

Comment (grant-Kolena) [reply kolena/_experimental/dataset/_evaluation.py]: yes, this is better

File: kolena/_experimental/dataset/_evaluation.py [hunk 7]
```diff
@@ -162,52 +139,28 @@ def _validate_configs(configs: List[TYPE_EVALUATION_CONFIG]) -> None:
...
-        on = [on]
-
-    _validate_on(df_datapoints, df_result_input, on)
-    try:
-        df_result = df_datapoints[on].merge(df_result_input, how="left", on=on, validate="one_to_one")
-    except MergeError as e:
-        raise IncorrectUsageError(f"merge key {on} is not unique") from e
-
-    return df_result.drop(columns=on)
+def _validate_datapoint_id_columns(df: pd.DataFrame, datapoint_id_columns: List[str]) -> None:
```
Comment (ly-kolena) [inline kolena/_experimental/dataset/_evaluation.py]: can recently abstracted `validate_dataframe_ids` in common.py be reused here?

**Review by ly-kolena — APPROVED** [review]

File: kolena/_experimental/dataset/_evaluation.py (hunk 7 above)
Comment (grant-Kolena) [reply kolena/_experimental/dataset/_evaluation.py]: sure

**Pull request merged**